pos_product_no_merge/static/src/js/models.js

```diff
@@ -0,0 +1,27 @@
+/** ***************************************************************************
+    Copyright (C) 2023 - Today: GRAP (http://www.grap.coop)
+    @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+    License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+******************************************************************************/
+
+odoo.define('pos_product_no_merge.model', function (require) {
+    "use strict";
+
+    var models = require('point_of_sale.models');
+    var core = require('web.core');
+    var session = require('web.session');
+    var _t = core._t;
+
+    var moduleOrderlineParent = models.Orderline;
+    models.Orderline = models.Orderline.extend({
+
+        can_be_merged_with: function(orderline){
+            if (orderline.product.pos_no_merge) {
+                return false;
+            }
+            return moduleOrderlineParent.prototype.can_be_merged_with.apply(this, arguments);
+        },
+
+    });
```

Comment:
💡 I recommend this pattern to deal with "super". I find it easier to read, specially when there are multiple metohds being overriden:

```suggestion
    const OrderlineSuper = models.Orderline.prototype;
    models.Orderline = models.Orderline.extend({
        can_be_merged_with: function(orderline){
            if (orderline.product.pos_no_merge) {
                return false;
            }
            return OrderlineSuper.can_be_merged_with.apply(this, arguments);
        },
    });
```